app/controllers/api/openid_connect/authorizations_controller.rb

```diff
@@ -0,0 +1,106 @@
+module Api
+  module OpenidConnect
+    class AuthorizationsController < ApplicationController
+      rescue_from Rack::OAuth2::Server::Authorize::BadRequest do |e|
+        logger.info e.backtrace[0, 10].join("\n")
+        render json: {error: e.message || :error, status: e.status}
+      end
+
+      before_action :authenticate_user!
+
+      def new
+        request_authorization_consent_form
+      end
+
+      def create
+        restore_request_parameters
+        process_authorization_consent(params[:approve])
+      end
+
+      private
+
```

Comment:
Assignment Branch Condition size for save_request_parameters is too high. [16/15]